Rework README and move dev instructions to contributing (#481)
UCL-ARC/python-tooling: README.md

> package template (e.g., [`scikit-hep`](https://github.com/scikit-hep/cookie)),
> we recommend using their template instead of this one.

## Tutorial

Some quick instructions for using our template are below.

> We also have a pedagogical [tutorial](./tutorial.md) that has been given in a couple of workshops geared towards researchers at UCL.

Reviewer: ```suggestion
We also have a [tutorial](./tutorial.md) that has been presented in a couple of workshops aimed at researchers at UCL.
```

Author: I put in {pedagogical, geared at} because it was in the old sentence that followed.

Don't feel strongly but will leave it for Matt's opinion when he's back.

Reviewer 2: Also don't have strong opinions here so happy for David's suggestion to go in!

## Using our template

If you have [uv] installed, you can use our template with the following one-liner:

```sh
uvx cookiecutter gh:ucl-arc/python-tooling --checkout latest
```

> You can also [install cookiecutter].

Reviewer: ```suggestion
Alternatively you can [install cookiecutter].
```

> Alternatively you can [install cookiecutter].

Reviewer 2: ```suggestion
Alternatively you can [install cookiecutter] using pipx or pip.
```

> Do this if you don't use [uv], or if you're likely to want to use cookiecutter again.

Reviewer: This says "you can do this if you don't use uv", but then the command below uses `uv`. Perhaps just link to https://cookiecutter.readthedocs.io/en/stable/README.html#installation instead of offering extra commands?

Author: I do link to there. Line above ☝️ 
And I thought "Here's the way we install it" indicated that there are other ways.

Reviewer 2: Maybe we could change

> [install cookiecutter]

to

> [install cookiecutter] with pipx or pip

to make it clearer that this is distinct from the `uv` installation instructions and maybe also change

> Here's the way we install it:

to 

> To install cookiecutter using uv run

Author: How about just removing the whole thing and directing the user to install cookiecutter following cookiecutter's instructions?

https://github.com/UCL-ARC/python-tooling/pull/481/commits/6816c2e03d12b495c8c5410b12d4d2fa560beb67

Then it becomes a bit simpler.

Reviewer 2: Looks good to me!

> Here's the way we install it:

Reviewer 2: ```suggestion
To install cookiecutter using uv run:
```

  [3/14] Email address for package author - will be part of package metadata
(grace.hopper@ucl.ac.uk):
  [4/14] Name of project - may contain spaces (Python Template): Example Research Software Project
  [5/14] 'Slugified' project name for use in URLs - dash-case recommended
(example-research-software-project):
  [6/14] Name for Python package - snake_case recommended
(example_research_software_project): ersp
  [7/14] Short description of the project (A cookiecutter package with UCL ARC
recommendations.): An example project.
  [8/14] Initialise project directory as a Git repository? [y/n] (y): y
  [9/14] Automatically deploy HTML docs to GitHub Pages on pushes to main? [y/n]
(y): y
  [10/14] GitHub user or organization name which will be owner of repository
(grace-hopper): UCL-ARC
  [11/14] Minimum Python version supported by package
    1 - 3.11
    2 - 3.12
    3 - 3.13
    Choose from [1/2/3] (1):
  [12/14] Maximum Python version supported by package
    1 - 3.13
    2 - 3.12
    3 - 3.11
    Choose from [1/2/3] (1):
  [13/14] Which open-source license to release package under
    1 - MIT
    2 - BSD-3
    3 - GPL-3.0
    Choose from [1/2/3] (1): 1

>   [14/14] Organisation(s) to acknowledge for funding of project (optional) ():

Reviewer 3: Do we want this? It's yet another thing to maintain that won't be very DRY, in addition to #432

Author: Maybe not... I mean cookiecutter _is_ quite self-explanatory.

Reviewer: I'm 👍 to removing this

Reviewer 2: I would also be in favour of removing to avoid having to keep up to date

https://github.com/UCL-ARC/example-research-software-project/settings/actions

After the 'Documentation' workflow has successfully completed at least once you will also need to configure the repository to deploy a GitHub pages site from the content on the gh-pages branch by going to

https://github.com/UCL-ARC/example-research-software-project/settings/pages

and under 'Built and deployment' selecting 'Deploy from a branch' for the 'Source' drop-down and 'gh-pages' for the 'Branch' drop-down, leaving the branch path drop-down with its default value of '/ (root)'.
```

</details>

<br/>

This will create a project directory structure.

<details><summary>Click to expand...</summary>

```sh
tree example-research-software-project
example-research-software-project
├── CITATION.cff
├── LICENSE.md
├── README.md
├── docs
│   ├── LICENSE.md
│   ├── api.md
│   └── index.md
├── mkdocs.yml
├── pyproject.toml
├── schemas
│   └── github-issue-forms.json
├── src
│   └── ersp
│       └── __init__.py
└── tests
    └── test_dummy.py

6 directories, 11 files

> ```

Reviewer 3: Again some concern

Author: 🤷 

It's ~ what we had before. Happy to remove if consensus.

Reviewer 3: I feel it could get out of sync

Author: Let's see what @matt-graham says. But it looks like... two votes to remove these (?) ... which is compelling.

Reviewer 2: Also in favour of removing this from same perspective of avoiding having to keep this in sync
